orderapi/src/main/java/fi/hel/verkkokauppa/order/service/invoice/OrderItemInvoicingService.java

```diff
@@ -82,4 +83,45 @@ public List<OrderItemInvoicing> filterAndUpdateCancelledOrders(List<OrderItemInv
         }
         return filteredInvoicings;
     }
+
+    public Boolean canCancel(String orderId){
+        List<OrderItemInvoicing> orderItemInvoicings = orderItemInvoicingRepository.findByOrderId(orderId);
+
+        if( orderItemInvoicings == null || orderItemInvoicings.isEmpty() ){
+            // no order item invoicings, no reason why could not be cancelled
+            return true;
+        }
+
+        for (OrderItemInvoicing orderItemInvoicing : orderItemInvoicings) {
+            if( orderItemInvoicing.getStatus().equals(OrderItemInvoicingStatus.CREATED) ){
+                // there is something that can be cancelled
+                return true;
+            }
+        }
+        log.info("Order has nothing to cancel. OrderId: {}", orderId);
+        return false;
+    }
+
+    public void cancelOrderItemInvoicings(String orderId){
+        List<OrderItemInvoicing> orderItemInvoicings = orderItemInvoicingRepository.findByOrderId(orderId);
+
+        if( orderItemInvoicings == null || orderItemInvoicings.isEmpty() ){
+            // no order item invoicings to cancel
+            return;
+        }
+
+        log.info("Cancelling order item invoicings for order: {}", orderId);
+        for (OrderItemInvoicing orderItemInvoicing : orderItemInvoicings) {
+            if( orderItemInvoicing.getStatus().equals(OrderItemInvoicingStatus.INVOICED )){
```

Comment:
check that getStatus is not null and the orderItemInvoicing is not null before using it, add the check before this

Reply:
orderItemInvoicing.getStatus() != null && orderItemInvoicing.getStatus().equals(OrderItemInvoicingStatus.INVOICED